src/bp-document/classes/class-bp-rest-document-endpoint.php

```diff
@@ -2461,4 +2510,24 @@ protected function bbp_document_update_rest_field_callback( $object, $value ) {
 
 	}
 
+	/**
+	 * Get document id for the attachment.
+	 *
+	 * @param integer $attachment_id Attachment ID.
+	 *
+	 * @return array|bool
+	 */
+	public function bp_rest_attachment_document_id( $attachment_id = 0 ) {
+		global $bp, $wpdb;
+
+		if ( ! $attachment_id ) {
+			return false;
+		}
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.DirectDatabaseQuery
+		$attachment_document_id = (int) $wpdb->get_var( "SELECT DISTINCT d.id FROM {$bp->document->table_name} d WHERE d.attachment_id = {$attachment_id}" );
```

Comment:
@KartikSuthar Create a seprate function so if anyone need document id via the attachment id they can get and not makea seprate query everywhere.